skunkworks_crow/src/main/java/org/odk/share/tasks/DownloadJob.java

```diff
@@ -146,18 +188,28 @@ private DownloadEvent receiveForms() {
         return new DownloadEvent(DownloadEvent.Status.FINISHED, sbResult.toString());
     }
 
+    /**
+     * Close all the connections.
+     */
+    private void closeConnections() throws IOException {
+        if (socket != null) {
```

Comment:
No sure if closing socket before their dis and dos will hve any affect or not, but to be safe we can  close it after closing dis and dos.

Reply:
updated.